### PR TITLE
Blog Post: Announcing  the OTel Arrow Protocol

### PR DESCRIPTION
**Pull request by lquerel**
I'd like to publish a blog post to announce the introduction of the OTel Arrow Protocol and the corresponding experimental collector. 

Thank you in advance for your feedback.

---

**Preview**: https://deploy-preview-3436--opentelemetry.netlify.app/blog/2023/otel-arrow/

### CODE REVIEW

File: content/en/blog/2023/otel-arrow/index.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+of OTel receiver/exporter that supports this protocol is available in the
+[contrib repository](https://github.com/open-telemetry/opentelemetry-collector-contrib).
+
+## How can I leverage OTel Arrow in my deployment?
+
+In the initial phase of this project, our primary goal is to optimize
+communication between two collectors. This is commonly observed in setups where
+telemetry traffic is funneled through one or multiple collectors before being
+relayed across the internet for backend processing. Given the increased
+complexity of the OTel Arrow Protocol compared to the original OTLP Protocol,
```
Comment (svrnm) [inline content/en/blog/2023/otel-arrow/index.md:122]:
```suggestion
complexity of the OTel Arrow Protocol compared to the original OTLP,
```

File: content/en/blog/2023/otel-arrow/index.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+As is often the case, there isn't a one-size-fits-all solution. Deployments with
+limited resources or those generating minimal telemetry should stick with the
+standard collector based on OTLP. Moreover, OTel Arrow necessitates the support
+of bi-directional gRPC streams and some degree of batching to fully benefit from
+the columnar representation. This can make the solution unsuitable in certain
+specific scenarios. It's also worth noting that, during this phase of the
+project, a slight increase in CPU and memory usage is expected in the collector.
+This is due to the overhead of automatically translating OTLP objects to OTel
+Arrow objects. However, this overhead will be completely eliminated in the
```
Comment (svrnm) [inline content/en/blog/2023/otel-arrow/index.md:141]:
```suggestion
This is due to the overhead of automatically translating OTLP objects to OTel
Arrow Protocol objects. However, this overhead will be completely eliminated in the
```

**Review by svrnm — CHANGES_REQUESTED** [review]
Thank you for this blog post, this is an interesting read and some exciting news condensed in a very consumable way! Looking forward to have this published.

I did a round of reviews, please take a look. 

A thing that should be fixed and consistent throughout the document is the **official name** of the Protocol, it looks like it is "OTel Arrow Protocol" or is it only "OTel Arrow", or would "OpenTelemetry Arrow Protocol" (similar to OTLP = OpenTelemetry Protocol) be a choice as well? Please make also sure that the name complies with the ASF's trademark guidelines (https://www.apache.org/foundation/marks/)

File: content/en/blog/2023/otel-arrow/index.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+majority of workloads and even greater enhancements for workloads that contain
+multivariate metrics, which share attributes. One of the highlights of the OTel
+Arrow integration is its seamless adaptability. In a typical deployment,
+incorporating OTel Arrow doesn't necessitate any substantial changes. Users
+simply need to redeploy a new version of the collector with a slightly tweaked
+configuration.
+
+This new protocol will serve as the cornerstone for future advancements in
+telemetry data processing and will generally foster enhanced integration with
+modern telemetry backends.
```
[the remark this answers is not in the document]
Comment (lquerel) [reply content/en/blog/2023/otel-arrow/index.md:29]: Good point. I will change the intro as follow:

We are thrilled to unveil the **OTel Arrow Protocol**, which is based on [Apache Arrow](https://arrow.apache.org/), a columnar-oriented memory format used for developing analytics applications. We are also excited to announce the release of a new pair of receiver/exporter in the [opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/) repository that supports this protocol.

File: content/en/blog/2023/otel-arrow/index.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+The advancements don't end there. In a subsequent phase of this project, we aim
+to leverage the columnar layout to significantly enhance data processing speeds
+within an expanded OTel Collector architecture that natively supports a new
+Arrow-based pipeline. Based on our proof of concept, we anticipate at least an
+order of magnitude improvement in data processing speed with this updated
+collector.
+
+These different sources of inefficiency and misalignment are the rationale
+behind our support for the new OTel Arrow Protocol as an alternative to the
+existing OTLP protocol. Our decision to leverage the well known Apache Arrow
```
[the remark this answers is not in the document]
Comment (lquerel) [reply content/en/blog/2023/otel-arrow/index.md:101]: Done

File: content/en/blog/2023/otel-arrow/index.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+  specifically for consuming and/or producing OTel Arrow messages. By
+  streamlining communication between components, we expect to enhance data
+  processing efficiency. Significant acceleration is anticipated as conversions
+  to and from OTLP, as well as the serialization and deserialization of
+  telemetry batches, will be bypassed.
+- **Leveraging SIMD-based data processing engines from the Apache Arrow
+  ecosystem**: This will further accelerate telemetry data processing and expand
+  the range of data processing capabilities.
+- **Consideration of adding Parquet exporters**: Made possible by the existing
+  bridge between Arrow and Parquet.
```
[the remark this answers is not in the document]
Comment (lquerel) [reply content/en/blog/2023/otel-arrow/index.md:198]: Done

**Comment by lquerel** [comment]
@svrnm Thank you for your feedback. I believe I've made all the changes and implemented the suggestions you mentioned.

**Comment by svrnm** [comment]
> @svrnm Thank you for your feedback. I believe I've made all the changes and implemented the suggestions you mentioned.

There are some spelling issues left, please take a look, you can add an ignore list for words in the front matters:

```
cSpell:ignore: <words-you-want-to-ignore>
```

File: content/en/blog/2023/otel-arrow/index.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
+---
+title: Reduce Telemetry Traffic by Up to 2 Times Using OTel Arrow Protocol
+linkTitle: OTel Arrow Protocol
+date: 2023-09-14
```
Comment (chalin) [inline content/en/blog/2023/otel-arrow/index.md:4]: Let's bring the date a little closer to now :)

```suggestion
date: 2023-11-01
```

File: content/en/blog/2023/otel-arrow/index.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+author: >-
+  [Laurent Quérel](https://github.com/lquerel) (F5), [Joshua
+  MacDonald](https://github.com/jmacd) (ServiceNow)
+---
```
Comment (chalin) [inline content/en/blog/2023/otel-arrow/index.md:8]: This should fix the cspell warning.

```suggestion
cSpell:ignore: Quérel
---
```

File: content/en/blog/2023/otel-arrow/index.md [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+
+## Why a new protocol?
+
+The growth in telemetry data is undeniable and rapid. This surge is attributed
+to several factors:
+
+- the proliferation of devices and sensors,
+- the shift from monolithic application deployment to more granular forms like
+  containers and serverless functions,
+- and the increasing reliance on data-driven and AI-driven technologies.
```
Comment (chalin) [inline content/en/blog/2023/otel-arrow/index.md:43]:
```suggestion
- Proliferation of devices and sensors
- Shift from monolithic application deployment to more granular forms like
  containers and serverless functions
- Increasing reliance on data-driven and AI-driven technologies
```

File: content/en/blog/2023/otel-arrow/index.md [hunk 9]
```diff
@@ -0,0 +1,237 @@
+---
+title: Reduce Telemetry Traffic by Up to 2 Times Using OTel Arrow Protocol
```
Comment (chalin) [inline content/en/blog/2023/otel-arrow/index.md:2]: "Reduce" and "up to 2 times" doesn't flow well. How about a shorter title like:

```suggestion
title: Reduce Telemetry Traffic by Half Using the OTel Arrow Protocol
```

**Review by chalin — COMMENTED** [review]
Please rebase since there are conflicts.

A few drive-by copyedit comments.

**Comment by svrnm** [comment]
> A thing that should be fixed and consistent throughout the document is the **official name** of the Protocol, it looks like it is "OTel Arrow Protocol" or is it only "OTel Arrow", or would "OpenTelemetry Arrow Protocol" (similar to OTLP = OpenTelemetry Protocol) be a choice as well? Please make also sure that the name complies with the ASF's trademark guidelines ([apache.org/foundation/marks](https://www.apache.org/foundation/marks/))

@lquerel @jmacd I have to follow up and highlight this topic once again. Please take a look at [Project and Product Names Using “Apache Arrow”](https://arrow.apache.org/powered_by/):

> Names like “Apache Arrow BigCoProduct” are not OK, as are names including “Apache Arrow” in general. The above links, however, describe some exceptions, like for names such as “BigCoProduct, powered by Apache Arrow” or “BigCoProduct for Apache Arrow”.

Applying that logic, "OTel Arrow Protocol" is not OK, but a name (likely, IANAL) following the ASF's trademark guidelines could be "OpenTelemetry Protocol for Apache Arrow" or "OpenTelemetry Protocol, powered by Apache Arrow". Alternatively dropping the Arrow and replacing it with something else would work of course.

**Comment by tigrannajaryan** [comment]
>Applying that logic, "OTel Arrow Protocol" is not OK

From https://arrow.apache.org/powered_by/

```
It is common practice to create software identifiers (Maven coordinates, module names, etc.) like “arrow-foo”. These are permitted. 
```

This seems to suggest "otel arrow" as an identifier of a protocol is acceptable.

If you want to avoid the issue altogether maybe use a more generic name, e.g. "otel columnar protocol".

File: content/en/blog/2023/otel-arrow/index.md (hunk 9 above)
Comment (lquerel) [reply content/en/blog/2023/otel-arrow/index.md]: @jmacd and I ended up choosing a different title.

**Comment by lquerel** [comment]
@svrnm  I think I have made all the changes you wanted. 
@jmacd  and I have decided to change the official name of this project to be fully align with the Apache Arrow project guidelines. The new official name is "OpenTelemetry Protocol with Apache Arrow". 
The repo identifier is still otel-arrow, but that's okay according to Apache Arrow documentation (see @tigran's comment).

The build failed and I'm not exactly sure why. Your help would be appreciated to finalize the publication of this article.

File: content/en/blog/2023/otel-arrow/index.md [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+The advancements don't end there. In a subsequent phase of this project, we aim
+to leverage the columnar layout to significantly enhance data processing speeds
+within an expanded OpenTelemetry Collector architecture that natively supports a
+new Arrow-based pipeline. Based on our proof of concept, we anticipate at least
+an order of magnitude improvement in data processing speed with this updated
+collector.
+
+These different sources of inefficiency and misalignment are the rationale
+behind our support for the new OpenTelemetry Protocol with Apache Arrow as an alternative to the
+existing OTLP protocol. Our decision to leverage the well known Apache Arrow
```
Comment (svrnm) [inline content/en/blog/2023/otel-arrow/index.md:130]:
```suggestion
existing OpenTelemetry Protocol. Our decision to leverage the well known Apache Arrow
```

OTLP Protocol is doubling Protocol

File: content/en/blog/2023/otel-arrow/index.md [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+Your deployments and feedback will assist us in further strengthening this
+project.
+
+To facilitate this testing phase, we have also improved the file exporter to
+save and anonymize the data in a compressed format. We have developed tools to
+enable the analysis and replay of this anonymized traffic in order to identify
+either bugs or sources of inefficiency. This approach recently played an
+important role in improving the logic that defines the best Arrow schemas to use
+for a given traffic. During an experiment in a production environment, we
+noticed a degradation in the compression rate after 1 million spans. Upon
```
Comment (svrnm) [inline content/en/blog/2023/otel-arrow/index.md:213]:
```suggestion
noticed a degradation in the compression rate after one million spans. Upon
```

**Review by svrnm — CHANGES_REQUESTED** [review]
overall lgtm, please take a look at the reviews. When all is done we can take care of the breaking CI jobs and the themes/docsy change that shouldn't be here

**Comment by lquerel** [comment]
@svrnm Thanks again for your feedback, I applied all your suggestions.

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
🎉 Thank you @lquerel!

**Pull request merged**